src/SearchableEntity.php

```diff
@@ -58,20 +65,47 @@ private function setId()
         }
 
         if (1 == count($ids)) {
-            $this->id = reset($ids);
-        } else {
-            $objectID = '';
-            foreach ($ids as $key => $value) {
-                $objectID .= $key . '-' . $value . '__';
-            }
-
-            $this->id = rtrim($objectID, '_');
+            $id = \reset($ids);
+            $this->id = \is_scalar($id) ? (string)$id : $this->serializeValue($id);
```

Comment:
The `$id` may implement the `__toString()` method. We may want to check of that first before asking for the `serializeValue`. What do you think?

Reply:
I think serializeValue should always return an array since we use either `Serializer::normalize` or `Normalizer::toArray`.